### PR TITLE
feat: add publish action to cli-api

### PR DESCRIPTION
**Pull request by oljekechoro**

### CODE REVIEW

File: packages/cli-api/src/main/ts/executors/deprecate.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import { defaultRateLimit } from '../default'
+import { TDeprecationConfig } from '../interfaces'
+import { printResults, printResultsJson, withRateLimit } from '../utils'
+
+export const performDeprecation = async (
+  config: TDeprecationConfig,
+  customBatchClient?: INpmRegClientWrapper
+): Promise<void> => {
+  const regClient = new RegClient()
```
Comment (antongolub) [inline packages/cli-api/src/main/ts/executors/deprecate.ts:11]: There's no need to create a new client if `customBatchClient` is passed

File: packages/cli-api/src/main/ts/executors/deprecate.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    'Following packages are not deprecated due to errors:'
+  )
+}
+
+type TEnrichedBatchResult = {
+  result: any
+  packageInfo: IDeprecatePackageParams
+}
+
+export const enrichResults = (normalizedResults: any[], data: IDeprecatePackageParams[]): TEnrichedBatchResult[] =>
```
Comment (antongolub) [inline packages/cli-api/src/main/ts/executors/deprecate.ts:51]: I suggest to introduce smth like `INormalizedResult`

File: packages/cli-api/src/main/ts/executors/publish.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  const successfulPackages = enrichedResults
+    .filter((item) => item.result.status === 'fulfilled' && item.result.value.success)
+    .map(item => item.params)
+
+  const failedPackages = enrichedResults
+    .filter((item) => item.result.status === 'rejected' || !item.result.value.success)
+    .map((item: TEnrichedResult) => ({
+      ...item.params,
+      error: item.result.status === 'rejected' ? item.result.reason : 'no data'
```
Comment (antongolub) [inline packages/cli-api/src/main/ts/executors/publish.ts:47]: const values / status map ?

File: packages/cli-api/src/main/ts/utils/misc.ts [hunk 4]
```diff
@@ -1,3 +1,46 @@
+import assert from 'assert'
 import { readFileSync } from 'fs'
 
 export const readFileToString = (path: string): string => readFileSync(path).toString()
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const assertString = (value: any, name: string): void => {
```
Comment (antongolub) [inline packages/cli-api/src/main/ts/utils/misc.ts:7]: TODO migrate to `blork`
blork needs ts libdefs to be added at first

**Review by antongolub — CHANGES_REQUESTED** [review]

File: packages/cli-api/src/main/ts/executors/deprecate.ts (hunk 2 above)
Comment (oljekechoro) [reply packages/cli-api/src/main/ts/executors/deprecate.ts:46]: oh, it is old parameter name, I will rename it to `results`

**Review by antongolub — APPROVED** [review]

**Review by antongolub — APPROVED** [review]

**Pull request merged**